stream/distributedlog/core/src/main/java/org/apache/distributedlog/ReadAheadEntryReader.java

```diff
@@ -1014,6 +1014,9 @@ public void onSegmentsUpdated(List<LogSegmentMetadata> segments) {
         if (!started.get()) {
             return;
         }
+        if (segments == null || segments.isEmpty()) {
+            return;
+        }
         logger.info("segments is updated with {}", segments);
         processLogSegments(segments);
```

Comment:
If we returned when the segments is empty, it will skip call `processLogSegments(segments);`, and I'm not sure what's the impact

Reply:
processLogSegments seems to do nothing in case of empty segments List